src/mlpack/methods/edge_boxes/feature_extraction_impl.hpp

```diff
@@ -0,0 +1,1052 @@
+  {
+    f.subvec(0, Im.n_cols - 1) = Im.row(y).t();
+    arma::vec d;
+    this->DistanceTransform1D(f, Im.n_cols, inf, d);
+    Im.row(y) = d.t();
+  }
+}
+
+/**
+ * euclidean distance transform of binary Image using squared distance
+ * @param Im Input binary Image whose distance transform is to be found.
+ * @param on if on == 1, 1 is taken as boundaries and vice versa.
+ * @param Out Output Image.
+ */
+template<typename MatType, typename CubeType>
+void StructuredForests<MatType, CubeType>::
+DistanceTransformImage(const MatType& Im, double on, MatType& Out)
+{
+  //need a large value but not infinity.
+  double inf = 999999.99;
+  Out = MatType(Im.n_rows, Im.n_cols, arma::fill::zeros);
+  Out.elem( find(Im != on) ).fill(inf);
+  this->DistanceTransform2D(Out, inf);
+}
+
+/**
+ * Makes a reflective border around an Image.
+ * @param InImage, Image which we have to make border around.
+ * @param top, border length (to be incremented) at top.
+ * @param left, border length at left.
+ * @param bottom, border length at bottom.
+ * @param right, border length at right.
+ * @param OutImage, Output Image. 
+ */
+template<typename MatType, typename CubeType>
+void StructuredForests<MatType, CubeType>::
+CopyMakeBorder(const CubeType& InImage, size_t top, 
+               size_t left, size_t bottom, size_t right,
+               CubeType& OutImage)
+{
+  OutImage = CubeType(InImage.n_rows + top + bottom, InImage.n_cols + left + right, InImage.n_slices);
+
+  for(size_t i = 0; i < InImage.n_slices; ++i)
+  {
+    OutImage.slice(i).submat(top, left, InImage.n_rows + top - 1, InImage.n_cols + left - 1)
+     = InImage.slice(i);
+    
+    for(size_t j = 0; j < right; ++j)
+    {
+      OutImage.slice(i).col(InImage.n_cols + left + j).subvec(top, InImage.n_rows + top - 1)
+      = InImage.slice(i).col(InImage.n_cols - j - 1);  
+    }
+
+    for(size_t j = 0; j < left; ++j)
+    {
+      OutImage.slice(i).col(j).subvec(top, InImage.n_rows + top - 1)
+      = InImage.slice(i).col(left - 1 - j);
+    }
+
+    for(size_t j = 0; j < top; j++)
+    {
+
+      OutImage.slice(i).row(j)
+      = OutImage.slice(i).row(2 * top - 1 - j);
+    }
+     
+    for(size_t j = 0; j < bottom; j++)
+    {
+      OutImage.slice(i).row(InImage.n_rows + top + j)
+      = OutImage.slice(i).row(InImage.n_rows + top - j - 1);
+    }
+
+  }
+}
+
+/**
+ * Converts an Image in RGB color space to LUV color space.
+ * RGB must range in (0.0, 1.0).
+ * @param InImage Input Image in RGB color space.
+ * @param OutImage Ouptut Image in LUV color space.
+ */
+template<typename MatType, typename CubeType>
+void StructuredForests<MatType, CubeType>::
+RGB2LUV(const CubeType& InImage, CubeType& OutImage,\
+        const arma::vec& table)
+{
+  //assert type is double or float.
+
+
+  MatType rgb2xyz;
+  rgb2xyz << 0.430574 << 0.222015 << 0.020183 << arma::endr
+          << 0.341550 << 0.706655 << 0.129553 << arma::endr
+          << 0.178325 << 0.071330 << 0.939180;
```

Comment:
I am not sure what you mean by this here. We are  populating the rgb2xyz matrix with the corresponding values.